server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -207,6 +211,19 @@ public DocTableInfo(RelationName ident,
             .toList();
     }
 
+
+    /**
+     * <p>Contains all information to parse the source:</p>
+     * @param referencesByOid is needed to rewrite OID-s to column names.
+     * OID-s is casted to String since source keys are stored as String
+     * and it's cheaper to cast long->String once and do OID rewrites without per-row casting.
+     * @param droppedColumns is needed to hide dropped columns from the source.
+     * @param versionCreated is needed to ignore DROP COLUMN related logic for old tables.
+     */
+    public static record SourceParsingMetadata(Map<String, Reference> referencesByOid,
```

Comment:
Feels a bit odd to have `SourceParsing` components in the `DocTableInfo`. Breaks encapsulation.

The `DocTableInfo` should imho not know anything about the storage format, or what kind of information is required to parse it. I'd move this somewhere to the execution engine components

Reply:
After internal discussion, we removed it and passing a field name resolver instead